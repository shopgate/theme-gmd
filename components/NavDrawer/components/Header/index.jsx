/**
 * Copyright (c) 2017-present, Shopgate, Inc. All rights reserved.
 *
 * This source code is licensed under the Apache 2.0 license found in the
 * LICENSE file in the root directory of this source tree.
 */

import React from 'react';
import PropTypes from 'prop-types';
import I18n from '@shopgate/pwa-common/components/I18n';
import { LOGIN_PATH } from '@shopgate/pwa-common/constants/RoutePaths';
import AccountBoxIcon from 'Components/icons/AccountBoxIcon';
import styles from './style';
import Item from '../Item';

/**
 * The Header component.
 * @param {Object} props The component props.
 * @returns {JSX}
 */
const Header = (props) => {
  if (!props.user) {
    return (
      <div className={styles.container} data-test-id="NavDrawerLoginButton">
        <Item href={LOGIN_PATH} icon={AccountBoxIcon} close={props.close} primary>
          <I18n.Text string="navigation.login_register" />
        </Item>
      </div>
    );
  }

  return (
    <div className={`${styles.container} ${styles.loggedIn}`} data-test-id="NavDrawerLoginButton">
      <div className={styles.welcome} data-test-id="LoginWelcome">
        <I18n.Text string="navigation.welcome_message" params={{ name: props.user.firstName }} />
      </div>
      <div className={styles.mail}>
        {props.user.mail}
      </div>
    </div>
  );
};

Header.propTypes = {
  close: PropTypes.func,
  user: PropTypes.shape(),
};

Header.defaultProps = {
  close: () => {},
  user: null,
};

export default Header;
