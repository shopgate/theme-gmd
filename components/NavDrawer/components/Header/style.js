/**
 * Copyright (c) 2017-present, Shopgate, Inc. All rights reserved.
 *
 * This source code is licensed under the Apache 2.0 license found in the
 * LICENSE file in the root directory of this source tree.
 */

import { css } from 'glamor';
import colors from 'Styles/colors';
import variables from 'Styles/variables';

const container = css({
  color: colors.accentContrast,
  marginBottom: 4,
}).toString();

const loggedIn = css({
  background: colors.accent,
  display: 'flex',
  flexDirection: 'column',
  justifyContent: 'center',
  minHeight: variables.navigator.height,
  padding: `${variables.gap.small + 1}px ${variables.gap.big}px ${variables.gap.small - 1}px`,
}).toString();

const ellipsis = {
  textOverflow: 'ellipsis',
  overflow: 'hidden',
  lineHeight: 1.3,
};

const welcome = css({
  fontWeight: 500,
  ...ellipsis,
}).toString();

const mail = css({
  ...ellipsis,
}).toString();

export default {
  container,
  loggedIn,
  welcome,
  mail,
};
