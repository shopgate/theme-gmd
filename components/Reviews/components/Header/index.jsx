/**
 * Copyright (c) 2017-present, Shopgate, Inc. All rights reserved.
 *
 * This source code is licensed under the Apache 2.0 license found in the
 * LICENSE file in the root directory of this source tree.
 */

import React from 'react';
import PropTypes from 'prop-types';
import { RATING_SCALE_DIVISOR } from 'Components/RatingStars/constants';
import I18n from '@shopgate/pwa-common/components/I18n';
import AverageRating from './components/AverageRating';
import WriteReviewLink from './components/WriteReviewLink';
import styles from './style';

/**
 * The header of the reviews component
 * @param {Object} rating The rating values
 * @param {bool} withTopGap Adds additional top gap when true.
 * @returns {JSX}
 */
const Header = ({ rating, withTopGap }) => {
  if (!rating) {
    return null;
  }
  const { average = 0 } = rating;
  const containerClass = withTopGap ? styles.withTopGapContainer : styles.container;

  if (!average) {
    return (
      <div className={styles.container}>
        <AverageRating rating={rating} />
        <div className={styles.noReviews}>
          <I18n.Text string="reviews.no_reviews" />
          <WriteReviewLink />
        </div>
      </div>
    );
  }

  return (
    <div className={containerClass} id="reviewsExcerpt">
      <AverageRating rating={rating} />
      <div className={styles.reviewsLine}>
        <I18n.Text string="reviews.rating" className={styles.averageRatingText}>
          <span className={styles.averageRatingNumber}>
            {average / RATING_SCALE_DIVISOR}
          </span>
        </I18n.Text>
        <WriteReviewLink />
      </div>
    </div>
  );
};

Header.propTypes = {
  rating: PropTypes.shape(),
  withTopGap: PropTypes.bool,
};

Header.defaultProps = {
  rating: null,
  withTopGap: false,
};

export default Header;
