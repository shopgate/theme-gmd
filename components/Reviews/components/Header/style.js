/**
 * Copyright (c) 2017-present, Shopgate, Inc. All rights reserved.
 *
 * This source code is licensed under the Apache 2.0 license found in the
 * LICENSE file in the root directory of this source tree.
 */

import { css } from 'glamor';
import colors from 'Styles/colors';
import variables from 'Styles/variables';

const container = css({
  fontWeight: 500,
  margin: 0,
}).toString();

const withTopGapContainer = css({
  ...container,
  marginTop: variables.gap.xbig,
}).toString();

const reviewsLine = css({
  display: 'flex',
  justifyContent: 'space-between',
  alignItems: 'baseline',
  padding: `0 0 ${variables.gap.small}px`,
  marginBottom: -2,
}).toString();

const averageRatingNumber = css({
  color: colors.primary,
  marginLeft: variables.gap.small,
}).toString();

const averageRatingText = css({
  marginLeft: variables.gap.big,
}).toString();

const noReviews = css({
  display: 'flex',
  justifyContent: 'space-between',
  flexDirection: 'column',
  alignItems: 'center',
  marginTop: variables.gap.small,
}).toString();

export default {
  container,
  reviewsLine,
  averageRatingNumber,
  averageRatingText,
  withTopGapContainer,
  noReviews,
};
