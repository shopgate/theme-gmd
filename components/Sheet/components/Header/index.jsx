/**
 * Copyright (c) 2017-present, Shopgate, Inc. All rights reserved.
 *
 * This source code is licensed under the Apache 2.0 license found in the
 * LICENSE file in the root directory of this source tree.
 */

import React, { Component } from 'react';
import PropTypes from 'prop-types';
import Grid from '@shopgate/pwa-common/components/Grid';
import Ripple from 'Components/Ripple';
import CrossIcon from 'Components/icons/CrossIcon';
import styles from './style';

/**
 * Header component.
 */
class Header extends Component {
  static propTypes = {
    title: PropTypes.string.isRequired,
    onToggleClose: PropTypes.func,
  };

  /**
   * The component's default props.
   * @type {Object}
   */
  static defaultProps = {
    onToggleClose: () => {},
  };

  /**
   * Only re-render when the title changes.
   * @param {Object} nextProps Next Props
   * @returns {boolean}
   */
  shouldComponentUpdate(nextProps) {
    return (this.props.title !== nextProps.title);
  }

  /**
   * Renders the component.
   * @returns {JSX}
   */
  render() {
    return (
      <Grid component="div" wrap={false}>
        <button className={styles.closeButton} onClick={this.props.onToggleClose}>
          <Ripple className={styles.closeIcon}>
            <CrossIcon size={24} />
          </Ripple>
        </button>
        <Grid.Item className={styles.title} component="div" grow={1}>
          {this.props.title}
        </Grid.Item>
      </Grid>
    );
  }
}

export default Header;
