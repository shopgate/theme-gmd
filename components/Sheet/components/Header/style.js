/**
 * Copyright (c) 2017-present, Shopgate, Inc. All rights reserved.
 *
 * This source code is licensed under the Apache 2.0 license found in the
 * LICENSE file in the root directory of this source tree.
 */

import { css } from 'glamor';
import variables from 'Styles/variables';

const closeButton = css({
  lineHeight: 1,
  outline: 0,
  padding: 0,
  minWidth: variables.navigator.height,
  height: variables.navigator.height,
  position: 'relative',
  zIndex: 2,
}).toString();

const closeIcon = css({
  display: 'flex',
  height: '100%',
  width: '100%',
  justifyContent: 'center',
  alignItems: 'center',
}).toString();

const title = css({
  fontSize: '1.25rem',
  fontWeight: 500,
  position: 'relative',
  alignItems: 'center',
  padding: `0 ${variables.gap.big}px`,
  overflow: 'hidden',
  textOverflow: 'ellipsis',
  whiteSpace: 'nowrap',
  alignSelf: 'center',
}).toString();

export default {
  closeButton,
  closeIcon,
  title,
};
