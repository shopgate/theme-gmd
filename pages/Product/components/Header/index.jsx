/**
 * Copyright (c) 2017-present, Shopgate, Inc. All rights reserved.
 *
 * This source code is licensed under the Apache 2.0 license found in the
 * LICENSE file in the root directory of this source tree.
 */

import React from 'react';
import Grid from '@shopgate/pwa-common/components/Grid';
import CTAButtons from './components/CTAButtons';
import Rating from './components/Rating';
import Name from './components/Name';
import Manufacturer from './components/Manufacturer';
import PriceStriked from './components/PriceStriked';
import Shipping from './components/Shipping';
import Availability from './components/Availability';
import Price from './components/Price';
import PriceInfo from './components/PriceInfo';
import Tiers from './components/Tiers';
import styles from './style';

/**
 * The product header component that displays textual information
 * - manufacturer
 * - shipping
 * - discount
 * - price
 * - tier prices
 * - price info
 * If not available or animating it will display placeholders
 * @returns {JSX}
 */
const ProductHeader = () => (
  <div className={styles.content}>
    <CTAButtons />
    <Rating />
    <Name />
    <Grid component="div">
      <Grid.Item component="div" grow={1}>
        <div className={styles.productInfo}>
          <Manufacturer />
        </div>
        <div className={styles.productInfo}>
          <Shipping />
        </div>
        <div className={styles.productInfo}>
          <Availability />
        </div>
      </Grid.Item>
      <Grid.Item component="div" className={styles.priceContainer}>
        <div className={styles.priceInfo}>
          <PriceStriked />
        </div>
        <div className={styles.priceInfo}>
          <Price />
        </div>
        <div className={styles.priceInfo}>
          <PriceInfo />
        </div>
        <div className={styles.priceInfo}>
          <Tiers />
        </div>
      </Grid.Item>
    </Grid>
  </div>
);

export default ProductHeader;
