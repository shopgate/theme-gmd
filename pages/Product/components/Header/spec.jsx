import React from 'react';
import { mount } from 'enzyme';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { Provider } from 'react-redux';
import mockRenderOptions from '@shopgate/pwa-common/helpers/mocks/mockRenderOptions';
import { defaultContext } from './../../__mocks__/context';
import { basicProductState } from './../mock';

const mockedStore = configureStore([thunk]);

jest.mock('Components/Reviews/components/Header', () => () => <div />);

jest.mock('./../../context');

describe.skip('<ProductHeader>', () => {
  /**
   * @param {Object} state A redux store.
   * @returns {JSX}
   */
  const createComponent = (state) => {
    // eslint-disable-next-line global-require
    const ProductHeader = require('./index').default;
    const store = mockedStore(state);
    return mount(
      <Provider store={store}>
        <ProductHeader productId={defaultContext.productId} />
      </Provider>,
      mockRenderOptions
    );
  };

  const components = [
    'CTAButtons',
    'Manufacturer',
    'Availability',
    'Rating',
    'Name',
    'Price',
    'PriceStriked',
    'Shipping',
    'Tiers',
  ];

  beforeEach(() => {
    jest.resetModules();
  });

  it('should render', () => {
    const { productId } = basicProductState.product.currentProduct;
    defaultContext.productId = productId;
    const cmp = createComponent(basicProductState);
    expect(cmp).toMatchSnapshot();
    components.forEach((c) => {
      expect(cmp.find(c).exists()).toBe(true);
    });
  });
});
