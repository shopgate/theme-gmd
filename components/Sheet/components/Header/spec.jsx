/**
 * Copyright (c) 2017-present, Shopgate, Inc. All rights reserved.
 *
 * This source code is licensed under the Apache 2.0 license found in the
 * LICENSE file in the root directory of this source tree.
 */

import React from 'react';
import { shallow } from 'enzyme';
import Header from './index';

describe('<Header />', () => {
  it('should render with correct title', () => {
    const title = 'My Title';
    const wrapper = shallow(
      <Header title={title} />
    );

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('GridItem').first().props().children).toEqual(title);
  });
});
