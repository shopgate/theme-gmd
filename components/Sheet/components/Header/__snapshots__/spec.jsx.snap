// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> should render with correct title 1`] = `
<Grid
  className=""
  component="div"
  wrap={false}
>
  <button
    className="css-b3dcfs"
    onClick={[Function]}
  >
    <Ripple
      className="css-ugy6xl"
      color="#000"
      disabled={false}
      fill={false}
      onComplete={[Function]}
      overflow={false}
      size={null}
    >
      <Cross
        size={24}
      />
    </Ripple>
  </button>
  <GridItem
    className="css-1mjy0de"
    component="div"
    grow={1}
    shrink={1}
  >
    My Title
  </GridItem>
</Grid>
`;
